tests/aws/test_serverless.py

```diff
@@ -43,22 +47,40 @@ def setup_and_teardown(self, aws_client, region_name, delenv):
         existing_api_ids = [api["id"] for api in apis]
 
         # deploy serverless app
-        run(
-            ["npm", "run", "deploy", "--", f"--region={region_name}"],
-            cwd=base_dir,
-            env_vars={"AWS_ACCESS_KEY_ID": TEST_AWS_ACCESS_KEY_ID},
-        )
+        if is_aws_cloud():
+            run(
+                ["npm", "run", "deploy-aws", "--", f"--region={region_name}"],
+                cwd=base_dir,
+            )
+        else:
+            run(
+                ["npm", "run", "deploy", "--", f"--region={region_name}"],
+                cwd=base_dir,
+                env_vars={"AWS_ACCESS_KEY_ID": TEST_AWS_ACCESS_KEY_ID},
+            )
 
         yield existing_api_ids
 
         try:
+            # cleanup s3 bucket content
+            bucket_name = f"testing-bucket-sls-test-{get_deployed_stage}-{region_name}"
+            response = aws_client.s3.list_objects_v2(Bucket=bucket_name)
+            objects = [{"Key": obj["Key"]} for obj in response.get("Contents", [])]
+            if objects:
+                aws_client.s3.delete_objects(
+                    Bucket=bucket_name,
+                    Delete={"Objects": objects},
+                )
             # TODO the cleanup still fails due to inability to find ECR service in community
-            run(["npm", "run", "undeploy", "--", f"--region={region_name}"], cwd=base_dir)
+            if is_aws_cloud():
+                run(["npm", "run", "undeploy-aws", "--", f"--region={region_name}"], cwd=base_dir)
+            else:
+                run(["npm", "run", "undeploy", "--", f"--region={region_name}"], cwd=base_dir)
```

Comment:
nit: maybe a personal preference, we can write something like the following since only one argument is changing in the above case: 

```
command = "undeploy-aws" if is_aws_cloud() else "undeploy"
run(["npm", "run", command, "--", f"--region={region_name}"], cwd=base_dir)
```
or maybe also try running: 

`run(["npm", "run", "undeploy-aws" if is_aws_cloud() else "undeploy", "--", f"--region={region_name}"], cwd=base_dir)`